api/src/tests/conftest.py

```diff
@@ -76,10 +76,50 @@ async def request_body():
     return request
 
 
-def pytest_addoption(parser):
-    parser.addoption("--integration", action="store_true", help="run integration tests")
+def pytest_configure(config: pytest.Config):
+    """Add markers to be recognised by pytest."""
+    has_unit_option = config.getoption("unit", default=False)
+    has_integration_option = config.getoption("integration", default=False)
+    marker_expr = config.getoption("markexpr", default="")
+    if marker_expr != "" and (has_integration_option or has_unit_option):
+        pytest.exit("Invalid options: Cannot use --markexpr with --unit or --integration options", 4)
 
 
-def pytest_runtest_setup(item):
-    if "integration" in item.keywords and not item.config.getoption("integration"):
-        pytest.skip("need --integration option to run")
+def pytest_addoption(parser):
+    """Add option to pytest parser for running unit/integration tests."""
+    parser.addoption("--unit", action="store_true", default=False, help="run unit tests")
+    parser.addoption("--integration", action="store_true", default=False, help="run integration tests")
+
+
+def pytest_collection_modifyitems(config: pytest.Config, items: list[pytest.Item]):
+    """Add markers to tests based on folder structure."""
+    unit_tests_directory = config.rootpath / "src/tests/unit"
+    integration_tests_directory = config.rootpath / "src/tests/integration"
+    for item in items:
+        if item.path.is_relative_to(unit_tests_directory):
+            item.add_marker("unit")
+        if item.path.is_relative_to(integration_tests_directory):
+            item.add_marker("integration")
+
+
+def pytest_runtest_setup(item: pytest.Item):
+    """Skip tests based on options provided."""
+    has_unit_option = item.config.getoption("unit", default=False)
+    has_integration_option = item.config.getoption("integration", default=False)
+    marker_expr = item.config.getoption("markexpr", default="")
+    match (has_unit_option, has_integration_option):
+        case (True, True):
+            # run all tests
+            return
+        case (False, True):
+            # skip unit tests
+            if "unit" in item.keywords:
+                pytest.skip("unit tests are skipped when explicitly running integration tests")
+        case (True, False):
+            # skip integration tests
+            if "integration" in item.keywords:
+                pytest.skip("integration tests are skipped when explicitly running unit tests")
+        case (False, False):
+            # skip integration tests
+            if "integration" in item.keywords and "integration" not in marker_expr:
+                pytest.skip("need --integration option to run")
```

Comment:
So if I don't have `--unit` or `--integration`, it will only run unit-tests? Why aren't all tests run?